### ISSUE TITLE
Misbehaviour when piping into the Ammonite REPL

### ISSUE THREAD

**Issue by milessabin** [lihaoyi/Ammonite#397]
I noticed this when attempting,
```
# yolo ...
curl -s https://raw.githubusercontent.com/milessabin/shapeless/master/scripts/try-shapeless.sh | bash
```
The REPL is downloaded and launched, however garbage appears on at the REPL prompt.

The following is a smaller reproduction,
```
haoyi-mbp:~ haoyi$ echo "println(1)" | ~/amm
Loading...
Welcome to the Ammonite Repl 0.5.8
(Scala 2.11.8 Java 1.8.0_25)
haoyi-haoyi@ println(1)
1

haoyi-haoyi@ ￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿￿...
```
See also #367.

**Comment by lihaoyi** [comment]
#367 is different - that's about using Ammonite's internal shellout syntax and letting it pipe external programs into each other, while this one is Ammonite itself behaving properly when piped into. 

Both should be fixed...

**Comment by coderabhishek** [comment]
I will try to fix this bug.

**Comment by blendmaster** [comment]
I'm getting this same behavior when feeding STDIN programmatically. If it helps debugging, I captured the actual output through `xxd`, which shows this odd expanding pattern of `efbf bfef` mixed with some terminal escapes.

https://paste.ubuntu.com/23220692/

**Comment by lihaoyi** [comment]
This is fixed in the latest unstable version `COMMIT-67a4081` http://www.lihaoyi.com/Ammonite/#UnstableVersions, thanks to @tkroman!
